src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -520,12 +520,14 @@ def _bid_aggregator(self, arguments: Dict):
 
             assert self.can_bid_be_posted(self.next_market.time_slot, **arguments)
 
-            replace_existing = arguments.get("replace_existing", True)
+            replace_existing = arguments.pop("replace_existing", True)
             bid = self.post_bid(
                 self.next_market,
                 arguments["price"],
                 arguments["energy"],
-                replace_existing=replace_existing)
+                replace_existing=replace_existing,
```

Comment:
So, we apparently dont have an integrationtest set up in the d3a-api-client that tests the requirements and attributes feature right? Can you please add a story for that and put it into the sprint?

Reply:
https://gridsingularity.atlassian.net/browse/D3ASIM-3554, Yes I will start with it just after merging this